simulator-ui/tests/entities.spec.ts

```diff
@@ -0,0 +1,206 @@
+      { testId: 'scenarioActionEntitiesStartDate', expectedValue: exampleDate },
+      { testId: 'scenarioActionEntitiesEndDate', expectedValue: exampleDate },
+    ],
+    testIdToBeVisible: [],
+  },
+  {
+    testName: 'should display table of test results and refresh button should work',
+    apiUrl: '**/api/test-results*',
+    entityUrl: 'http://localhost:9000/test-result',
+    contentJson: testResultJson,
+    locators: [
+      'th :text("Status")',
+      'th :text("Test Name")',
+      'th :text("Class Name")',
+      'th :text("Error Message")',
+      'th :text("Stack Trace")',
+    ],
+    testIdsAndExpectedValues: [
+      { testId: 'testResultEntitiesId', expectedValue: '1' },
+      { testId: 'testResultEntitiesStatus', expectedValue: 'FAILURE' },
+      { testId: 'testResultEntitiesTestName', expectedValue: 'Scenario(Default)' },
+      { testId: 'testResultEntitiesClassName', expectedValue: 'DefaultTestCase' },
+      { testId: 'testResultEntitiesErrorMessage', expectedValue: 'New Error' },
+      { testId: 'testResultEntitiesStackTrace', expectedValue: 'New Stacktrace' },
+      { testId: 'testResultEntitiesFailureType', expectedValue: '' },
+      { testId: 'testResultEntitiesCreatedDate', expectedValue: exampleDate },
+      { testId: 'testResultEntitiesLastModifiedDate', expectedValue: exampleDate },
+    ],
+    testIdToBeVisible: [],
+  },
+  {
+    testName: 'should display table of test parameters and refresh button should work',
+    apiUrl: '**/api/test-parameters*',
+    entityUrl: 'http://localhost:9000/test-parameter',
+    contentJson: testParameterJson,
+    locators: [
+      'th :text("Key")',
+      'th :text("Value")',
+      'th :text("Test Result")',
+      'th :text("Created Date")',
+      'th :text("Last Modified Date")',
+    ],
+    testIdsAndExpectedValues: [
+      { testId: 'testParameterEntityKey', expectedValue: 'test key' },
+      { testId: 'testParameterEntityValue', expectedValue: 'test value' },
+      { testId: 'testParameterEntityTestResultLink', expectedValue: '0' },
+      { testId: 'testParameterEntityCreatedDate', expectedValue: exampleDate },
+      { testId: 'testParameterEntityLastModifiedDate', expectedValue: exampleDate },
+    ],
+    testIdToBeVisible: ['', ''],
+  },
+];
+
+entityPageContentMap.forEach((contentObject: EntityPageContentObject) => {
+  test(`${contentObject.testName}`, async ({ page }) => {
+    // 'first test'
+    await mockBackendResponse(page, contentObject.apiUrl, contentObject.contentJson);
+
+    await page.goto(contentObject.entityUrl);
+
+    await expect(page.locator('th :text("ID")').nth(0)).toHaveCount(1);
+    await checkEntityPageContentValueAndVisibility(page, contentObject);
+    // 'second test'
+    await checkIfRefreshButtonWorks(page, contentObject);
+  });
+});
+
+test('should show message headers when clicking button on message row', async ({ page }) => {
```

Comment:
I would use `describe` for grouping or "specializing" tests.. e.g. here I would do something like this:

```suggestion
test.describe('message entity table', () => {
    test('should show message headers when clicking button', async ({ page }) => {
```

if I understood the test correctly, of course. see [docs](https://playwright.dev/docs/api/class-test#test-describe) for more details.

Reply:
adapted